extensions/amp-form/0.1/amp-form.js

```diff
@@ -177,6 +192,73 @@ export class AmpForm {
 
 
 /**
+ * Reports validity of the form passed through state object.
+ * @param {!Object} state
+ */
+function reportValidity(state) {
+  reportFormValidity(state.form);
+}
+
+
+/**
+ * Reports validity for the first invalid input - if any.
+ * @param {!HTMLFormElement} form
+ */
+function reportFormValidity(form) {
+  const inputs = toArray(form.querySelectorAll('input,select,textarea'));
+  for (let i = 0; i < inputs.length; i++) {
+    if (!inputs[i].checkValidity()) {
+      reportInputValidity(inputs[i]);
+      break;
+    }
+  }
+}
+
+
+/**
+ * Revalidates the currently focused input after a change.
+ * @param {!KeyboardEvent} event
+ */
+function onInvalidInputKeyUp_(event) {
+  if (event.target.checkValidity()) {
+    validationBubble.hide();
```

Comment:
One last: We may end up double calling `#hide` and `#show`. Storing the current message as a state on `ValidationBubble`, we can check if we're already hidden, or if we're trying to show the same message again. In either case, we can return without doing anything.